frontend/docsettings.lua

```diff
@@ -154,6 +174,20 @@ function DocSettings:getFileFromHistory(hist_name)
     end
 end
 
+--- Returns the directory and full filepath of a hash-ID-based sidecar metadata store
+-- @string doc_path path to the document (e.g., `/foo/bar.pdf`)
+function DocSettings:getSidecarHashDirAndFilepath(doc_path)
+    -- Getting PDF ID from trailer via mupdf has not been implemented - everything uses partial MD5
+    local path = self:getSidecarDir(doc_path, "hash") -- could this be cached somehow?
```

Comment:
I do think so (haven't yet read the whole thing, so may be it's already done).
I think that here or elsewhere, you should cache in a simple module local table `filepath => partialMD5`, so at least, when you browse directories and go up and down in a same KOReader session, you don't need to recompute the md5, and you can just pick it from that table. It's trashed when you exit KOReader, but it's ok.